learnCSharp/Commands.cs

```diff
@@ -45,35 +84,68 @@ public async static void RemoveTodo(ITodoRepository todoRepository, String[] inp
                 String secondaryInput = Console.ReadLine();
                 await todoRepository.DeleteTodo(int.Parse(secondaryInput));
             }
+            return;
         }
 
 
-        public static String ChangeUser(String[] input)
+        public async static Task<User> ChangeUser(String[] input, IUserRepository userRepository)
         {
-            String user;
+            User user;
+            String uName;
             if (input.Length > 1)
             {
-                user = input[1];
+                uName = input[1];
+                user = await userRepository.GetUserByName(input[1]);
             }
             else
             {
                 Console.WriteLine("ENTER TASK OWNER");
-                user = Console.ReadLine();
+                uName = Console.ReadLine();
+                user = await userRepository.GetUserByName(uName);
+            }
+            if (user == null)
+            {
+                user = new User() { Name = uName };
+                Console.WriteLine("New User Detected.\nPlease Enter Password for New User.");
+                user.Password = Console.ReadLine();
+                return await userRepository.NewUser(user);
+            }
+            else
+            {
+                Console.WriteLine("Please Enter Your Password:");
+                if(user.Password == Console.ReadLine())
+                {
+                    Console.WriteLine("accepted");
+                    return user;
+                }
+                else
+                {
+                    Console.WriteLine("Password Not Accepted");
+                    Console.WriteLine("Enter User Name");
+                    if(input.Length > 1)
+                    {
+                        input[1] = Console.ReadLine();
+                    }
+                    else
+                    {
+                        input.Append<String>(Console.ReadLine());
+                    }
```

Comment:
Nitpick (no need to change): Rather than have to do this logic, you can always just create a new array.